sale_commission_partial_settlement/wizard/wizard_settle.py

```diff
@@ -0,0 +1,101 @@
+        settlement_obj = self.env["sale.commission.settlement"]
+        settlement_line_obj = self.env["sale.commission.settlement.line"]
+        settlement_ids = []
+
+        if self.agent_ids:
+            agents = self.agent_ids
+        else:
+            agents = self.env["res.partner"].search([("agent", "=", True)])
+        date_to = self.date_to
+        for agent in agents:
+            date_to_agent = self._get_period_start(agent, date_to)
+            partial_agent_lines, agent_lines_to_update = self.get_partial_agent_lines(
+                agent, date_to_agent, self.date_payment_to
+            )
+            for line_id in agent_lines_to_update:
+                self.env["account.invoice.line.agent"].browse(line_id).update(
+                    agent_lines_to_update[line_id]
+                )
+            for company in partial_agent_lines.mapped(
+                "invoice_line_agent_id.company_id"
+            ):
+                agent_lines_company = partial_agent_lines.filtered(
+                    lambda r: r.invoice_line_agent_id.object_id.company_id == company
+                )
+                pos = 0
+                sett_to = date(year=1900, month=1, day=1)
+                while pos < len(agent_lines_company):
+                    line = agent_lines_company[pos]
+                    pos += 1
+
+                    if line.invoice_line_agent_id.invoice_date > sett_to:
+                        sett_from = self._get_period_start(
+                            agent, line.invoice_line_agent_id.invoice_date
+                        )
+                        sett_to = self._get_next_period_date(
+                            agent,
+                            sett_from,
+                        ) - relativedelta(days=1)
+                        settlement = self._get_settlement(
+                            agent, company, sett_from, sett_to
+                        )
+                        if not settlement:
+                            settlement = settlement_obj.create(
+                                self._prepare_settlement_vals(
+                                    agent, company, sett_from, sett_to
+                                )
+                            )
+                        settlement_ids.append(settlement.id)
+                    settlement_line_obj.create(
+                        {
+                            "settlement_id": settlement.id,
+                            "agent_line_partial_ids": [(6, 0, [line.id])],
+                            "agent_line": [(6, 0, [line.invoice_line_agent_id.id])],
+                        }
+                    )
+        if len(settlement_ids):
+            return {
+                "name": _("Created Settlements"),
+                "type": "ir.actions.act_window",
+                "views": [[False, "list"], [False, "form"]],
+                "res_model": "sale.commission.settlement",
+                "domain": [["id", "in", settlement_ids]],
+            }
```

Comment:
**quibble(non-blocking)**: can perhaps be slightly cleaner, but this is just my preference:

```suggestion
            return {
                "name": _("Created Settlements"),
                "type": "ir.actions.act_window",
                "view_type": "form",
                "view_mode": "tree,form",
                "res_model": "sale.commission.settlement",
                "domain": [("id", "in", settlement_ids)],
            }
```